core/src/main/scala/org/apache/spark/deploy/mesos/MesosDriverDescription.scala

```diff
@@ -40,24 +41,28 @@ private[spark] class MesosDriverDescription(
     val cores: Double,
     val supervise: Boolean,
     val command: Command,
-    val schedulerProperties: Map[String, String],
+    _schedulerProperties: Map[String, String],
     val submissionId: String,
     val submissionDate: Date,
     val retryState: Option[MesosClusterRetryState] = None)
   extends Serializable {
 
+  val sc = new SparkConf(false)
+  _schedulerProperties.foreach {case (k, v) => sc.set(k, v)}
+
   def copy(
       name: String = name,
       jarUrl: String = jarUrl,
       mem: Int = mem,
       cores: Double = cores,
       supervise: Boolean = supervise,
       command: Command = command,
-      schedulerProperties: Map[String, String] = schedulerProperties,
+      schedulerProperties: SparkConf = sc,
```

Comment:
This is more semantically correct, and helps with some consistency issues in the docker methods